### DISCUSSION TITLE
Add giftingMeta.ranksGiven to hypixel:player_info

### DISCUSSION THREAD

**Discussion by imyanice** [HypixelDev/ModAPI#36] [Ideas and Feedback]
Hello,
I think that adding giftingMeta.ranksGiven to the hypixel:player_info packet would be great and useful.
Thanks for taking this suggestion in consideration

**Comment by mdashlw** [comment]
It'd be useful to have a lot of things in `hypixel:player_info`. I don't think it's feasible to keep adding new stuff to it.

A potential solution would be a new subscription-only `hypixel:player_data` packet with json data from the db. It'd be a lot of data, but the idea is that if some mod subscribes to it, it'd have requested it via API anyway.  It could also make some mods a lot simpler if that's all the data they need. The question is if it should be login-only or requested at will.

**Comment by ConnorLinfoot** [comment]
As we've already pointed out we do not intend on the Mod API being used as a replacement for the Public API. The Mod API is intended for data about the currently active situation of the player/client, and where we think it is a benefit to communicate that to mods. 

I'm unsure the reason why mods would need this sort of context, especially often enough that it would warrant it being in the Mod API. But regardless, we currently have no plans to expose data such as ranks given via the Mod API.
